### PR TITLE
Created/Update Components for Make a Post

### PR DESCRIPTION
**Pull request by Will-Hsu**
Alert: customized alert component for Homehub.
CheckBoxGroup: customized checkbox component for Homehub.
TextArea: has a placeholder field that is grey out in addition to pre-filled content.
Tooltip: background-color responsive to the boolean prop isSingleLine.

All 4 components' storybooks are implemented as well.

### CODE REVIEW

File: FrontEnd/src/components/basics/Alert/Alert.module.scss [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  flex-direction: column;
+  flex-wrap: wrap;
+  margin: 0rem 0.75rem;
+  margin-left: 2.5rem;
+  margin-right: 2.5rem;
+}
+
+.title {
+  font-weight: 600;
+  font-size: 20px;
```
Comment (Yizong98) [inline FrontEnd/src/components/basics/Alert/Alert.module.scss:32]: rem

File: FrontEnd/src/components/basics/Alert/Alert.module.scss [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  line-height: 120%;
+  letter-spacing: 0.015em;
```
Comment (Yizong98) [inline FrontEnd/src/components/basics/Alert/Alert.module.scss:34]: why are we mixing em and rem here?

File: FrontEnd/src/components/basics/Alert/Alert.module.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  .button {
+    @include svg-fill(#f3d990, 1.5rem, 1.5rem);
+  }
+}
+
+.info {
+  background: #e2e6fa;
+  color: #6d83e7;
+  .button {
+    @include svg-fill(#6d83e7, 1.5rem, 1.5rem);
```
Comment (Yizong98) [inline FrontEnd/src/components/basics/Alert/Alert.module.scss:76]: 1.5 rem is default, no need to specify

File: FrontEnd/src/components/basics/Alert/Alert.module.scss [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  .button {
+    @include svg-fill(#e16161, 1.5rem, 1.5rem);
+  }
+}
+
+.warning {
+  background: #fffaea;
+  color: #202020;
+  .button {
+    @include svg-fill(#f3d990, 1.5rem, 1.5rem);
```
Comment (Yizong98) [inline FrontEnd/src/components/basics/Alert/Alert.module.scss:68]: same here

File: FrontEnd/src/components/basics/Alert/Alert.module.scss [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  .button {
+    @include svg-fill(#94b36c, 1.5rem, 1.5rem);
+  }
+}
+
+.alert {
+  background: #f9dfdf;
+  color: #e16161;
+  .button {
+    @include svg-fill(#e16161, 1.5rem, 1.5rem);
```
Comment (Yizong98) [inline FrontEnd/src/components/basics/Alert/Alert.module.scss:60]: same here

File: FrontEnd/src/components/basics/Alert/Alert.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+export type alertVariant = 'success' | 'alert' | 'warning' | 'info';
+
+interface AlertProps {
+  variant: alertVariant;
+  title?: string;
+  text: string;
+  button?: string;
+}
+
+function getIcon(variant: string): string {
```
Comment (Yizong98) [inline FrontEnd/src/components/basics/Alert/Alert.tsx:16]: it is recommended to use an object: e.g. const IconMapping = {'success':alertIcons.Check, ..... }

File: FrontEnd/src/components/basics/Tooltip/Tooltip.module.scss [hunk 7]
```diff
@@ -5,12 +5,23 @@
   border-radius: 0.5rem;
   border: 1px solid $border-color;
   padding: 0.75rem;
+  box-shadow: 0px 1px 8px rgba(0, 0, 0, 0.18);
```
Comment (Yizong98) [inline FrontEnd/src/components/basics/Tooltip/Tooltip.module.scss:8]: same here, it needs to be rem

**Review by Yizong98 — CHANGES_REQUESTED** [review]
Extremely good job! Well done. Just need some minor fixes before I give you the approval. Thanls

File: FrontEnd/src/components/basics/Tooltip/Tooltip.module.scss (hunk 7 above)
Comment (Will-Hsu) [reply FrontEnd/src/components/basics/Tooltip/Tooltip.module.scss]: fixed

File: FrontEnd/src/components/basics/Alert/Alert.tsx (hunk 6 above)
Comment (Will-Hsu) [reply FrontEnd/src/components/basics/Alert/Alert.tsx]: fixed

File: FrontEnd/src/components/basics/Alert/Alert.module.scss (hunk 5 above)
Comment (Will-Hsu) [reply FrontEnd/src/components/basics/Alert/Alert.module.scss]: fixed

File: FrontEnd/src/components/basics/Alert/Alert.module.scss (hunk 4 above)
Comment (Will-Hsu) [reply FrontEnd/src/components/basics/Alert/Alert.module.scss]: fixed

File: FrontEnd/src/components/basics/Alert/Alert.module.scss (hunk 3 above)
Comment (Will-Hsu) [reply FrontEnd/src/components/basics/Alert/Alert.module.scss]: fixed

File: FrontEnd/src/components/basics/Alert/Alert.module.scss (hunk 2 above)
Comment (Will-Hsu) [reply FrontEnd/src/components/basics/Alert/Alert.module.scss]: fixed

File: FrontEnd/src/components/basics/Alert/Alert.module.scss (hunk 1 above)
Comment (Will-Hsu) [reply FrontEnd/src/components/basics/Alert/Alert.module.scss]: fixed

**Pull request merged**